tests/core/pyspec/eth2spec/test/context.py

```diff
@@ -365,6 +393,45 @@ def wrapper(*args, spec: Spec, **kw):
     return decorator
 
 
+def with_config_overrides(config_overrides):
+    """
+    WARNING: the spec_test decorator must wrap this, to ensure the decorated test actually runs.
+    This decorator forces the test to yield, and pytest doesn't run generator tests, and instead silently passes it.
+    Use 'spec_configured_state_test' instead of 'spec_state_test' if you are unsure.
+
+    This is a decorator that applies a dict of config value overrides to the spec during execution.
+    """
+    def decorator(fn):
+        def wrapper(*args, spec: Spec, **kw):
+            # remember the old config
+            old_config = spec.config
+
+            # apply our overrides to a copy of it, and apply it to the spec
+            tmp_config = deepcopy(old_config._asdict())  # not a private method, there are multiple
+            tmp_config.update(config_overrides)
+            config_types = spec.Configuration.__annotations__
+            # Retain types of all config values
+            test_config = {k: config_types[k](v) for k, v in tmp_config.items()}
+
+            # Output the config for test vectors  (TODO: check config YAML encoding)
+            yield 'config', 'data', test_config
```

Comment:
Ah, so you read `config.yaml` for configuration overrides in test vectors?

Reply:
Yes, just read the runtime config (if any is there) from the test case dir, and good to go. This should be the same in all test vectors, so it makes sense to yield it from within the config decorator.